lib/streams/documentStream.js

```diff
@@ -5,6 +5,16 @@ const through = require( 'through2' );
 const peliasModel = require( 'pelias-model' );
 const NAME_REGEX = /^name(_json)?_([a-z]{2})$/i;
 
+function getPopularity(record) {
+  const popularityString = getCaseInsensitive('popularity', record);
+  const popularity = parseInt(popularityString);
+  if (popularityString && (isNaN(popularity) || !/^\d+$/.test(popularityString))) {
+    throw new Error("Popularity must be an int, got " + popularityString)
```

Comment:
I don't think that we should throw errors for popularity.... This is not a required field, so a default value is enough when the data is incorrect.

Reply:
I can't recall, does `throw`-ing here kill the whole importer or does it print a message and skip just this one row?

In general, we've had this problem across all the importers, where different people have one of two needs:
- they want the import to finish if at all possible, even if one file/record/etc fails
- they want the import to be as _complete_ as possible, and would rather the entire import fails than some invalid data makes its way in, or some data is missed

Where needed, we've introduced flags like [`missingFilesAreFatal`](https://github.com/pelias/openaddresses/#importsopenaddressesmissingfilesarefatal) in the OA importer to toggle between the two preferences.

So, if this `throw` causes the whole importer to fail, then we need to provide a way to allow invalid records to be skipped. If it prints a warning/error and continues on importing other records, then it's probably fine.

Reply:
Okay, looks like it does just skip that one record. Now you all know the background anyway :)